### PR TITLE
feat: :sparkles: add a bypass fn in opts

### PR DESCRIPTION
**Pull request by ArnaudTA**
As requested in #51 and cause I need it too.

Here is my proposal, a bypass function can be send in options, the function receive the request and return a boolean, if true prevent the authentication methods, if false let the flow goes as usual.

Let me know what you think before I start writing tests.

### CODE REVIEW

**Comment by yubinTW** [comment]
Hi, thanks for your PR.
Could you add the document on the Configuration section in `READMD.md`? and add some test cases for it?

**Comment by ArnaudTA** [comment]
done but I can run test on my computer and I still don't know why, can you give a try to the CI ?

**Comment by yubinTW** [comment]
hey, I found the cause, since `Task` is lazy Promise, you have to add `()` to execute it

please refer the code below (the second line from the bottom)

```typescript
fastify.addHook('preValidation', (request: FastifyRequest, reply: FastifyReply, done) => {
    pipe(
      request,
      bypassFn,
      TO.match(
        () => {
          done()
        },
        (request) =>
          pipe(
            request,
            filterGrantRoute,
            O.chain(filterExcludedPattern),
            O.match(
              () => {
                done()
              },
              (request) =>
                pipe(
                  request,
                  getBearerTokenFromRequest,
                  O.match(
                    () => authenticationByGrant(request, reply, done),
                    (bearerToken) => authenticationByToken(request, reply, bearerToken, done)
                  )
                )
            )
          )
      )
    )() // <----- Here! 
  })
```

File: test/server.test.ts [hunk 1]
```diff
@@ -167,4 +167,97 @@ describe('server with keycloak testing', () => {
...
+    it('should return 401, cause bypassFn returned false', async () => {
+      const response = await server.inject({
+        method: 'GET',
+        url: '/ping',
+        headers: {
+          authorization: `Bearer fakeToken`,
+          password: 'mellon'
+        }
+      })
+      expect(response.statusCode).toBe(403)
```
Comment (yubinTW) [inline test/server.test.ts:259]: the assertion would be `401`

File: src/keycloak.ts [hunk 2]
```diff
@@ -450,21 +463,30 @@ export default fastifyPlugin(async (fastify: FastifyInstance, opts: KeycloakOpti
...
+                getBearerTokenFromRequest,
+                O.match(
+                  () => authenticationByGrant(request, reply, done),
+                  (bearerToken) => authenticationByToken(request, reply, bearerToken, done)
+                )
+              )
           )
+        )
       )
     )
```
Comment (yubinTW) [inline src/keycloak.ts:491]: need to add a `()`

**Comment by ArnaudTA** [comment]
thanks for your help fp-ts totally lost me

It works well now !

**Comment by yubinTW** [comment]
Update to v2.2.0, Thanks your PR !

**Pull request merged**